src/guess.py

```diff
@@ -202,30 +204,34 @@ def bayesdb_guess_stattypes(column_names, rows, null_values=None,
     for ci, column_name in enumerate(column_names):
         if casefold(column_name) in override_map:
             stattype = override_map[casefold(column_name)]
+            reason = 'User override.'
         else:
             column = nullify(null_values, rows, ci)
-            stattype = guess_column_stattype(column,
+            [stattype, reason] = guess_column_stattype(column,
                                              distinct_ratio=distinct_ratio,
                                              nullify_ratio=nullify_ratio,
                                              numcat_count=numcat_count,
                                              numcat_ratio=numcat_ratio,
                                              have_key=(key is not None))
             if stattype == 'key':
                 key = column_name
-        stattypes.append(stattype)
+        stattypes.append([stattype, reason])
     return stattypes
 
-def guess_column_stattype(column, **kwargs):
+def guess_column_stattype(column, reason='', **kwargs):
```

Comment:
Right, as noted in some other review comment which appears to have disappeared into github's `/dev/null`... I later noticed the recursive call and was confused about why it was necessary.